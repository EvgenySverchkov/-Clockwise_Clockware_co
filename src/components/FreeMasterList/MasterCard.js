import React from "react";
import PropTypes from "prop-types";

import RadioBtn from "../FormComponents/RadioBtn";

const MasterCard = ({ id, name, rating, towns }) => (
  <div className="col-sm-4 mb-2 col-sm-6 col-lg-4">
    <div className="card">
      <div className="card-body">
        <h5 className="card-title">{name}</h5>
        <p className="card-text">Rating: {rating}</p>
        <p className="card-text">Towns: {towns}</p>
        <RadioBtn id={id} value={id} name={"chooseMaster"}>
          <label className="form-check-label" htmlFor={id}>
            Choose
          </label>
        </RadioBtn>
      </div>
    </div>
  </div>
);

MasterCard.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string.isRequired,
  rating: PropTypes.number.isRequired,
  towns: PropTypes.string.isRequired,
};

export default MasterCard;
