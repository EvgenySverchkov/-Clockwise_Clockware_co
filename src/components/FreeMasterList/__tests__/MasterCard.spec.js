import React from "react";
import {create} from "react-test-renderer";

import MasterCard from "../MasterCard";

it("Test of <MasterCard /> for <FreeMasterList/>", ()=>{
    const mockProps = {
        id: 1, 
        name: "Name", 
        rating: 5
    }
    const component = create(<MasterCard {...mockProps}/>);

    expect(component.toJSON()).toMatchSnapshot();
});