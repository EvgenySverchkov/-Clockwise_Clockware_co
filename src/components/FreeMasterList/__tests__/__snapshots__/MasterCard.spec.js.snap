// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of <MasterCard /> for <FreeMasterList/> 1`] = `
<div
  className="col-sm-4 mb-2 col-sm-6 col-lg-4"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Name
      </h5>
      <p
        className="card-text"
      >
        Rating: 
        5
      </p>
      <div
        className="form-check form-check-inline"
      >
        <input
          className="form-check-input"
          id={1}
          name="chooseMaster"
          onChange={null}
          type="radio"
          value={1}
        />
        <label
          className="form-check-label"
          htmlFor={1}
        >
          Choose
        </label>
      </div>
    </div>
  </div>
</div>
`;
